tests/test_core.py

```diff
@@ -8229,6 +8229,42 @@ def test_pthread_dynamic_linking(self):
     self.set_setting('EXIT_RUNTIME')
     self.do_basic_dylink_test()
 
+  @needs_dlfcn
+  @node_pthreads
+  def test_Module_dynamicLibraries_pthreads(self):
+    # test that Module.dynamicLibraries works with pthreads
+
+    self.emcc_args += ['-pthread', '-Wno-experimental']
+    self.set_setting('PROXY_TO_PTHREAD')
+    self.set_setting('EXIT_RUNTIME')
+    self.set_setting('MODULARIZE')
+    self.set_setting('EXPORT_NAME', 'createMyModule')
+
+    def post(filename):
+      shutil.move(filename, '{filename}.module.js')
+      create_test_file(filename, '''
+        var createMyModule = require('./{filename}.module.js');
```

Comment:
I am not completely sure how robust this is.
It assumes that filename is a relative path and that the test runs in node (or an environment with `require`).
Seems to work fine locally.

Reply:
Is "{filename}" supposed to be a sting substitution in python here?   If you use would need to use the `f` prefix on the string..

Reply:
Yes, thanks!
Python is not my main (or non-main) language

Reply:
I think you could alternatively using `--extern-pre-js` and `--extern-post-js` which might be cleaner.

Reply:
Great, that sounds much neater.